ansible/roles/backup/tasks/main.yml

```diff
@@ -0,0 +1,149 @@
+---
+# Stop services
+- name: Stop js-ceramic-4 node
+  ansible.builtin.systemd:
+    name: js-ceramic
+    state: stopped
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+
+- name: Stop go-ipfs Docker container
+  ansible.builtin.shell:
+    cmd: docker-compose down
+    chdir: /opt/go-ipfs
+  when: inventory_hostname == 'gitcoin-go-ipfs-2.3boxlabs.com'
+  register: docker_compose_result
+
+# Ceramic backup
+- name: Get current timestamp
+  ansible.builtin.set_fact:
+    timestamp: "{{ ansible_date_time.iso8601_basic_short }}"
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+
+- name: Backup Ceramic daemon config file with timestamp
+  ansible.builtin.copy:
+    src: "{{ data_store_mount_path }}/daemon.config.json"
+    dest: "~/ceramic_config_backup_{{ timestamp }}"
+    remote_src: yes
+  when: inventory_hostname == 'gitcoin-js-ceramic-4.3boxlabs.com'
+
+- name: Get latest common snapshot between gitcoin-js-ceramic-3 and gitcoin-js-ceramic-4
+  ansible.builtin.shell:
+    cmd: |
+      local_snapshots=$(zfs list -H -t snapshot -o name ceramiconepool/data-store)
+      remote_snapshots=$(gcloud compute ssh gitcoin-js-ceramic-3 --zone "us-central1-a" --command="zfs list -H -t snapshot -o name ceramiconepool/data-store")
```

Comment:
sshing from one host to another? Or is this happening locally?

In either case, I think there's an easier way. A host task can access the vars of another host

- The syntax is hostvars['hostname']['variable_name'].
- Make sure the play that sets the fact runs before the play that uses it.
- The fact must be set in a previous task or play for this to work

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_vars_facts.html#information-about-ansible-magic-variables

Reply:
This was to get the list of ZFS snapshots from the other node so we can transfer just the latest incremental snapshot over.